core/commands/resolve.go

```diff
@@ -82,12 +82,15 @@ Resolve the value of an IPFS DAG path:
 		// the case when ipns is resolved step by step
 		if strings.HasPrefix(name, "/ipns/") && !recursive {
 			p, err := n.Namesys.ResolveN(req.Context(), name, 1)
+
+			if p != "" {
```

Comment:
I don't like it too but in other case there is no reason to have resolve without recursion.

You can't use it to do step by step resolving as you just get error of recursion exceeded.
I will remove this from the PR as this is something that might require some refactoring to get right.